### PR TITLE
post: cli 6.0

### PR DESCRIPTION
**Pull request by cexbrayat**
WIP

- [x] breaking change NodeJS 8.9
- [x] breaking change bundle names 
- [x] breaking change ng 5.x and 6.x only
- [x] change post date
- [x] add link to ng6 post
- [x] Webpack 4
- [x] dynamic lazy-loading
- [x] zone-error in env
- [x] architecture
- [x] ng update
- [x] mutli apps/libs support

### CODE REVIEW

File: _posts/2018-03-28-angular-cli-6.0.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Something like:
+
+    constructor(loader: SystemJsNgModuleLoader) {
+      loader.load('app/admin/admin.module#AdminModule')
+        .then(factory => ...);
+    }
+
+The problem was that the CLI was only able to bundle modules separately
+if they are found in a `loadChildren` route configuration.
+So you had to "trick" the CLI and Webapck to build a separate chunk.
```
Comment (clacote) [inline _posts/2018-03-28-angular-cli-6.0.md]: Webapck -> Webpack

File: _posts/2018-03-28-angular-cli-6.0.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Now that the unpleasant stuff is out of the way, let's see what this new release brings.
+
+## Webpack 4
+
+You probably know that under the hood the CLI uses Webpack to do the heavy lifting.
+Webpack has released the 4.0 version: you can read more about it on https://medium.com/webpack/webpack-4-released-today-6cdb994702d4.
+
+TL;DR: Webpack 4 is faster, should be smarter for bundling common parts of the application,
+a new option (`sideEffects`) will help to have a better tree-shaking, and adds WebAssembly support.
```
Comment (jnizet) [inline _posts/2018-03-28-angular-cli-6.0.md]: *has* a new option (`sideEffects`) *that* will help

File: _posts/2018-03-28-angular-cli-6.0.md [hunk 3]
```diff
@@ -9,24 +9,145 @@ description: "Angular CLI 6.0 is out! Which new features are included? Webpack 4
...
+[Angular Packaging Format](https://docs.google.com/document/d/1CZC2rcpxffTDfRDs6p1cfbmKNLA6x5O-NtkJglDaBVs/edit).
+Based on `ng-packagr`, the CLI can now build a library, and produces all the required files
+(es5 bundle, esm2015 bundle, umd bundle, metadata file for AoT compilation, public API file...).
+I'm not an expert on the topic,
+but it looks like you just need to `npm publish` the result and you're good to go!
+
+You can also have several applications in your project, with `ng generate application`.
+Actually you already have two now by default: your main application and an application containing the `e2e` tests.
+
+The cool thing is that you can directly import from the library into your applications.
```
Comment (clacote) [inline _posts/2018-03-28-angular-cli-6.0.md]: into your applications... in the same source code repository, right? That's not clear to me, maybe with NPM publication that could be used anywhere, but I think what your describing later will only work for the application in the same folder.

File: _posts/2018-03-28-angular-cli-6.0.md [hunk 4]
```diff
@@ -9,24 +9,145 @@ description: "Angular CLI 6.0 is out! Which new features are included? Webpack 4
...
+But a few commands are a special case and can be run directly, like `build`, `lint`, `test`, `xi18n`. `ng serve` and `ng e2e` needs the project to be specified, except if there is just one with this target in the workspace.
+
+So running `ng build` is the same as running `ng run *:build`,
+`ng lint my-app` is the same as running `ng run my-app:lint`,
+`ng serve` is the same as running `ng run my-app:serve`,
+you get it...
+
+A few commands are not delegating to `@angular-devkit/architect` but to `@angular-devkit/schematics`.
+These commands are `ng new my-app` (which is the same as `ng generate @schematics/angular:application my-app`), `ng update` and `ng add`.
+But I'll come back to these two last in a dedicated section.
```
Comment (clacote) [inline _posts/2018-03-28-angular-cli-6.0.md]: shouldn't "these two last" be in plural form "lasts"? Not sure, for such an adjective used as noun.

**Comment by cexbrayat** [comment]
Last commit makes the blog post almost ready (just need the correct dates and links). Up for review when you have time.

**Comment by jnizet** [comment]
ng-bootstrap has that, too (but not in master yet, AFAIK). I saw it mentioned in the ng-conf keynote, and checked if it was BS or not. And it's not, apparently: https://github.com/ng-bootstrap/schematics

**Comment by cexbrayat** [comment]
Yeah but there is just a README in master and a PR in progress (which did not work when I tried it), so I omitted it :)

**Comment by cexbrayat** [comment]
2 small commits for cli 6 blog post to review when you have time please 🙏

File: _posts/2018-03-28-angular-cli-6.0.md [hunk 5]
```diff
@@ -222,6 +227,44 @@ The configuration files and the project layout have changed quite a bit,
 as we pointed out above, so you'll have to move things around and migrate your configuration files
 (with `ng update` and/or manually by checking [`angular-cli-diff`](https://github.com/cexbrayat/angular-cli-diff/compare/1.7.0...6.0.0))
 
+Note that the environment concept has slightly changed
+and is now called a `configuration`.
```
Comment (jnizet) [inline _posts/2018-03-28-angular-cli-6.0.md]: Is that really true? It feels like they've just introduced a new concept (configuration), which allows applying an environment and choosing build options. The environment concept stays the same, doesn't it?

Comment (cexbrayat) [reply _posts/2018-03-28-angular-cli-6.0.md]: 👍 I tried to clarify in a new commit.

**Comment by clacote** [comment]
Date changed to 2018-05-05 (tomorrow) and link to ng 6 post #256.

**Comment by clacote** [comment]
rebased onto `master`.

**Pull request merged**